### PR TITLE
Applies tax to userReceipt

### PR DESCRIPTION
**Pull request by jmtucker6**

### CODE REVIEW

File: SplitIt/app/src/main/java/project/cs495/splitit/MainActivity.java [hunk 1]
```diff
@@ -280,6 +280,8 @@ private String parceScanResults(ScanResults brScanResults, String groupId) {
                 .setVendor(brScanResults.merchantName().value() == null ? "Unknown" : brScanResults.merchantName().value())
                 .setDatePurchased(brScanResults.receiptDate().value())
                 .setPrice(brScanResults.total().value())
+                .setSubtotal(brScanResults.subtotal().value())
+                .setTax(brScanResults.taxes().value())
```
Comment (bpsokol) [inline SplitIt/app/src/main/java/project/cs495/splitit/MainActivity.java]: Get this error when adding a new receipt: "java.lang.NullPointerException: Attempt to invoke virtual method 'float com.microblink.FloatType.value()' on a null object reference
at project.cs495.splitit.MainActivity.parceScanResults(MainActivity.java:284)
at project.cs495.splitit.MainActivity.onActivityResult(MainActivity.java:268)"

File: SplitIt/app/src/main/java/project/cs495/splitit/ReceiptViewActivity.java [hunk 2]
```diff
@@ -408,15 +407,18 @@ private void createUserReceipt() {
...
-                        totalPrice += item.getPrice();
+                        subtotal += item.getPrice();
                     }
                 }
-                userReceipt.setPrice(totalPrice);
+                userReceipt.setSubtotal(subtotal);
+                float percentOfSubtotal = subtotal / receipt.getSubtotal();
+                userReceipt.setTax(receipt.getTax() * percentOfSubtotal);
+                userReceipt.setPrice(subtotal + userReceipt.getTax());
                 userReceipt.commitToDB(mDatabaseReference);
```
Comment (bpsokol) [inline SplitIt/app/src/main/java/project/cs495/splitit/ReceiptViewActivity.java]: Get this error when trying to generate a user receipt: "com.google.firebase.database.DatabaseException: Invalid value: Value cannot be NaN, Inf or -Inf. 
at project.cs495.splitit.models.UserReceipt.commitToDB(UserReceipt.java:21)
at project.cs495.splitit.ReceiptViewActivity$11.onDataChange(ReceiptViewActivity.java:422)"

Strangely it doesn't happen when I create a user receipt from the target receipt you shared with me, but it does with receipts I already had scanned.

Comment (jmtucker6) [reply SplitIt/app/src/main/java/project/cs495/splitit/ReceiptViewActivity.java]: I'm guessing it has to do with those prior receipts not having the tax info stored on them. I'll look into it

File: SplitIt/app/src/main/java/project/cs495/splitit/ReceiptViewActivity.java [hunk 3]
```diff
@@ -421,8 +420,7 @@ public void onDataChange(DataSnapshot dataSnapshot) {
                 if (masterReceiptSubtotal == 0) {
                     masterReceiptSubtotal = receipt.getPrice();
                 }
-                float percentOfSubtotal = subtotal / masterReceiptSubtotal;
-                userReceipt.setTax(masterReceiptTax * percentOfSubtotal);
+                userReceipt.setTax(receipt.getTax());
```
Comment (jmtucker6) [inline SplitIt/app/src/main/java/project/cs495/splitit/ReceiptViewActivity.java]: the tax stores the dollar amount, not the rate. Thus, we want to multiply that dollar amount by the portion of the subtotal accounted for by the assigned user. I think this should be reverted.

File: SplitIt/app/src/main/java/project/cs495/splitit/MainActivity.java [hunk 4]
```diff
@@ -290,7 +291,9 @@ private String parceScanResults(ScanResults brScanResults, String groupId) {
             item.addReceiptId(receiptId);
             item.commitToDB(database);
             receipt.addItem(item.getItemId());
+            subTotal += item.getPrice();
         }
+        if (receipt.getTax() == 0) receipt.setTax((receipt.getPrice() / subTotal) - 1);
```
Comment (jmtucker6) [inline SplitIt/app/src/main/java/project/cs495/splitit/MainActivity.java]: It shouldn't be necessary to get the rate. We can use the dollar amount of tax (receipt.getPrice() - subTotal)

**Pull request merged**